posthog/hogql/ast.py

```diff
@@ -361,7 +378,7 @@ def print_type(self) -> str:
 @dataclass(kw_only=True)
 class ArrayType(ConstantType):
     data_type: ConstantDataType = field(default="array", init=False)
-    item_type: ConstantType
+    item_type: Optional[ConstantType] = None  # none means any type
```

Comment:
do we separate between `any` and `UnknownType`? Should it just be the latter instead of None?

Reply:
We dont - will change to Unknown as default